crml/nft/src/lib.rs

```diff
@@ -974,6 +999,152 @@ decl_module! {
 				None => Err(Error::<T>::NotForFixedPriceSale.into()),
 			}
 		}
+
+		/// Create an offer on a token
+		/// Locks funds until offer is accepted, rejected or cancelled
+		#[weight = 0]
+		fn make_offer (
+			origin,
+			token_id: TokenId,
+			amount: Balance,
+			asset_id: AssetId,
```

Comment:
would recommend this become an enum so this api doesn't need to change when we add new offer types, also storage won't need migrating either e.g.
```rust
enum OfferType {
   Simple(AssetId, Balance)
}
```